canvas.go

```diff
@@ -48,8 +52,32 @@ type line struct {
 	p0, p1 image.Point
 }
 
+var (
+	pathSanitizer = regexp.MustCompile(`\W+`)
+	piSanitizer   = regexp.MustCompile(`<\?xml[^>]*>`)
```

Comment:
+1 to mhilton's suggestion of parsing the XML. That's the approach used here: https://github.com/ajstarks/svgo/blob/master/codepic/codepic.go

(I'd use a []byte type for Doc rather than string though)